packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -202,4 +202,12 @@ export class LocalChannelContext implements IChannelContext {
         assert(this.isLoaded && this.channel !== undefined, "Channel should be loaded to run GC");
         return this.channel.getGCData();
     }
+
+    public updateUsedRoutes(usedRoutes: string[]) {
```

Comment:
No change is required from this comment, just want to verify my understanding:

If I understood the flow correctly, we "compute" GC data which originates at the root of the tree, and then propagates down the nodes via `getGCData()` calls. This returns the information needed by the GC code. Then once that is complete, we do some computations on it at the root, and pass the results (used routes) back down the tree to each node via `updateUsedRoutes()` calls. Each node does some additional work to filter out only the used routes relevant to its children, and passes this data to its corresponding `SummarizerNode` to be used for diff checking and summarization.

Reply:
Yes, that is correct.
A slight change In this part:
`Each node does some additional work to filter out only the used routes relevant to its children, and passes this data to its corresponding SummarizerNode to be used for diff checking and and summarization.`
It first updates its SummarizerNode, then filters the routes relevant for its children and passes them to the children (handles delay-load case). This is important because a data store may contain used routes that are not for its children and should be considered in diff checking. For example - some blob or a sub-tree that is known only to the data store. We don't have anything like that now but its very much possible.